RequestService/RequestService.Core/Services/DailyDigestService.cs

```diff
@@ -0,0 +1,145 @@
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace RequestService.Core.Services
+{
+    public class DailyDigestService : IDailyDigestService
+    {
+        private readonly IUserService _userService;
+        private readonly IRepository _repository;
+        private readonly IOptionsSnapshot<ApplicationConfig> _applicationConfig;
+        private readonly ICommunicationService _communicationService;
+        private readonly ILogger<DailyDigestService> _logger;
+        private readonly IJobFilteringService _jobFilteringService;
+
+        public DailyDigestService(IUserService userService, IOptionsSnapshot<ApplicationConfig> applicationConfig,
+            ICommunicationService communicationService,
+            IRepository repository,
+            ILogger<DailyDigestService> logger,
+            IJobFilteringService jobFilteringService)
+        {
+            _userService = userService;
+            _repository = repository;
+            _communicationService = communicationService;
+            _applicationConfig = applicationConfig;
+            _logger = logger;
+            _jobFilteringService = jobFilteringService;
+        }
+
+
+        public async Task SendDailyDigestEmailAsync(CancellationToken cancellationToken)
+        {
+
+            var openRequests = _repository.GetOpenJobsSummaries();
+
+            if (openRequests.Count == 0)
+            {
+                _logger.LogWarning($"No Open Job Summaries Found when generating daily digest");
+                return;
+            }
+
+            var nationalSupportActivities = new List<SupportActivities>() { SupportActivities.FaceMask, SupportActivities.HomeworkSupport, SupportActivities.PhoneCalls_Anxious, SupportActivities.PhoneCalls_Friendly };
+
+            var users = await _userService.GetUsers(cancellationToken);
+
+            if (users == null || users.UserDetails == null || users.UserDetails.Count() == 0)
+            {
+                _logger.LogWarning($"No Users found when generating daily digest");
+                return;
+            }
+
+            users.UserDetails = users.UserDetails.Where(x => x.SupportRadiusMiles.HasValue);
+
+            foreach (var user in users.UserDetails)
+            {
+                try
+                {
+                    var activitySpecificSupportDistancesInMiles = nationalSupportActivities.Where(a => user.SupportActivities.Contains(a)).ToDictionary(a => a, a => (double?)null);
+
+                    var attachedDistances = await _jobFilteringService.FilterJobSummaries(openRequests, null, user.PostCode, _applicationConfig.Value.DistanceInMilesForDailyDigest, activitySpecificSupportDistancesInMiles, null, null, new List<JobStatuses>() { JobStatuses.Open }, cancellationToken);
+
+                    //if they dont have the community connector support activity, let remove any open requests in there.
+                    if (!user.SupportActivities.Contains(SupportActivities.CommunityConnector))
```

Comment:
This if block can go (providing we are filtering on group).

Reply:
done